test/testdata/namer/visibility.rb

```diff
@@ -53,3 +53,26 @@ def foo;
     private :foo
   end
 end
+
+class Foo5
+  T.unsafe(nil).private
+
+  def foo; end  # this does not end up being private
+end
+
+class Foo6
+  self.private
+
+  def foo; end
+end
+
+class NilClass
```

Comment:
Rather than monkey patching NilClass, let’s either add an `# error:` assertion capturing the fact that private doesn’t exist on nil, or initialize bar with T.unsafe(nil).

Reply:
I've felt weird of this part also. Thanks for your review. I added error an assertion on the test.